core/templates/domain/admin/admin-backend-api.service.spec.ts

```diff
@@ -1417,4 +1417,257 @@ describe('Admin backend api service', () => {
       'Config property does not exist.');
   }
   ));
+
+  // Tests for Admin Dev Mode Activities Tab.
+  it('should generate dummy explorations', fakeAsync(() => {
+    let action = 'generate_dummy_explorations';
+    let numDummyExpsToGenerate = 2;
+    let numDummyExpsToPublish = 1;
+    let payload = {
+      action: action,
+      num_dummy_exps_to_generate: numDummyExpsToGenerate,
+      num_dummy_exps_to_publish: numDummyExpsToPublish
+    };
+
+    abas.generateDummyExplorationsAsync(
+      numDummyExpsToGenerate, numDummyExpsToPublish
+    ).then(successHandler, failHandler);
+
+    let req = httpTestingController.expectOne(
+      '/adminhandler');
+    expect(req.request.method).toEqual('POST');
+    expect(req.request.body).toEqual(payload);
+    req.flush(200);
+    flushMicrotasks();
+
+    expect(successHandler).toHaveBeenCalled();
+    expect(failHandler).not.toHaveBeenCalled();
+  }
+  ));
+
+  it('should fail to generate dummy explorations', fakeAsync(() => {
```

Comment:
`should handle generate dummy explorations request failure` (Ditto elsewhere)

Reply:
Done